### ISSUE TITLE
Testa aren't passed

### ISSUE THREAD

**Issue by zdm** [maxmind/GeoIP2-perl#34]
```
--> Working on M/MA/MAXMIND/GeoIP2-2.003000.tar.gz
Fetching http://www.cpan.org/authors/id/M/MA/MAXMIND/GeoIP2-2.003000.tar.gz ... OK
GeoIP2-2.003000/
GeoIP2-2.003000/LICENSE
GeoIP2-2.003000/INSTALL
GeoIP2-2.003000/Makefile.PL
GeoIP2-2.003000/dist.ini
GeoIP2-2.003000/bin/
GeoIP2-2.003000/bin/web-service-request
GeoIP2-2.003000/perltidyrc
GeoIP2-2.003000/lib/
GeoIP2-2.003000/lib/GeoIP2.pm
GeoIP2-2.003000/lib/GeoIP2/
GeoIP2-2.003000/lib/GeoIP2/Record/
GeoIP2-2.003000/lib/GeoIP2/Record/Location.pm
GeoIP2-2.003000/lib/GeoIP2/Record/Country.pm
GeoIP2-2.003000/lib/GeoIP2/Record/Subdivision.pm
GeoIP2-2.003000/lib/GeoIP2/Record/Postal.pm
GeoIP2-2.003000/lib/GeoIP2/Record/Traits.pm
GeoIP2-2.003000/lib/GeoIP2/Record/City.pm
GeoIP2-2.003000/lib/GeoIP2/Record/Continent.pm
GeoIP2-2.003000/lib/GeoIP2/Record/RepresentedCountry.pm
GeoIP2-2.003000/lib/GeoIP2/Record/MaxMind.pm
GeoIP2-2.003000/lib/GeoIP2/Database/
GeoIP2-2.003000/lib/GeoIP2/Database/Reader.pm
GeoIP2-2.003000/lib/GeoIP2/WebService/
GeoIP2-2.003000/lib/GeoIP2/WebService/Client.pm
GeoIP2-2.003000/lib/GeoIP2/Types.pm
GeoIP2-2.003000/lib/GeoIP2/Error/
GeoIP2-2.003000/lib/GeoIP2/Error/Type.pm
GeoIP2-2.003000/lib/GeoIP2/Error/WebService.pm
GeoIP2-2.003000/lib/GeoIP2/Error/HTTP.pm
GeoIP2-2.003000/lib/GeoIP2/Error/IPAddressNotFound.pm
GeoIP2-2.003000/lib/GeoIP2/Error/Generic.pm
GeoIP2-2.003000/lib/GeoIP2/Model/
GeoIP2-2.003000/lib/GeoIP2/Model/Insights.pm
GeoIP2-2.003000/lib/GeoIP2/Model/ISP.pm
GeoIP2-2.003000/lib/GeoIP2/Model/AnonymousIP.pm
GeoIP2-2.003000/lib/GeoIP2/Model/Country.pm
GeoIP2-2.003000/lib/GeoIP2/Model/Domain.pm
GeoIP2-2.003000/lib/GeoIP2/Model/ConnectionType.pm
GeoIP2-2.003000/lib/GeoIP2/Model/Enterprise.pm
GeoIP2-2.003000/lib/GeoIP2/Model/City.pm
GeoIP2-2.003000/lib/GeoIP2/Role/
GeoIP2-2.003000/lib/GeoIP2/Role/Record/
GeoIP2-2.003000/lib/GeoIP2/Role/Record/Country.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Record/HasNames.pm
GeoIP2-2.003000/lib/GeoIP2/Role/HasIPAddress.pm
GeoIP2-2.003000/lib/GeoIP2/Role/HasLocales.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Model.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Error/
GeoIP2-2.003000/lib/GeoIP2/Role/Error/HTTP.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Model/
GeoIP2-2.003000/lib/GeoIP2/Role/Model/Location.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Model/HasSubdivisions.pm
GeoIP2-2.003000/lib/GeoIP2/Role/Model/Flat.pm
GeoIP2-2.003000/README.md
GeoIP2-2.003000/cpanfile
GeoIP2-2.003000/CONTRIBUTING.md
GeoIP2-2.003000/META.json
GeoIP2-2.003000/perlcriticrc
GeoIP2-2.003000/Changes
GeoIP2-2.003000/META.yml
GeoIP2-2.003000/tidyall.ini
GeoIP2-2.003000/MANIFEST
GeoIP2-2.003000/t/
GeoIP2-2.003000/t/00-report-prereqs.dd
GeoIP2-2.003000/t/author-eol.t
GeoIP2-2.003000/t/author-no-tabs.t
GeoIP2-2.003000/t/release-synopsis.t
GeoIP2-2.003000/t/lib/
GeoIP2-2.003000/t/lib/Test/
GeoIP2-2.003000/t/lib/Test/GeoIP2.pm
GeoIP2-2.003000/t/author-pod-spell.t
GeoIP2-2.003000/t/author-test-version.t
GeoIP2-2.003000/t/00-report-prereqs.t
GeoIP2-2.003000/t/release-portability.t
GeoIP2-2.003000/t/release-tidyall.t
GeoIP2-2.003000/t/author-pod-syntax.t
GeoIP2-2.003000/t/GeoIP2/
GeoIP2-2.003000/t/GeoIP2/Database/
GeoIP2-2.003000/t/GeoIP2/Database/Reader.t
GeoIP2-2.003000/t/GeoIP2/Database/Reader-Anonymous-IP.t
GeoIP2-2.003000/t/GeoIP2/Database/Reader-Enterprise.t
GeoIP2-2.003000/t/GeoIP2/Database/Reader-Connection-Type.t
GeoIP2-2.003000/t/GeoIP2/Database/Reader-Domain.t
GeoIP2-2.003000/t/GeoIP2/Database/Reader-ISP.t
GeoIP2-2.003000/t/GeoIP2/WebService/
GeoIP2-2.003000/t/GeoIP2/WebService/Client.t
GeoIP2-2.003000/t/GeoIP2/Error/
GeoIP2-2.003000/t/GeoIP2/Error/Type.t
GeoIP2-2.003000/t/GeoIP2/Types.t
GeoIP2-2.003000/t/GeoIP2/Model/
GeoIP2-2.003000/t/GeoIP2/Model/City.t
GeoIP2-2.003000/t/GeoIP2/Model/Insights.t
GeoIP2-2.003000/t/GeoIP2/Model/names.t
GeoIP2-2.003000/t/GeoIP2/Model/Country.t
GeoIP2-2.003000/t/release-pod-coverage.t
GeoIP2-2.003000/t/author-00-compile.t
GeoIP2-2.003000/t/release-cpan-changes.t
GeoIP2-2.003000/maxmind-db/
GeoIP2-2.003000/maxmind-db/LICENSE
GeoIP2-2.003000/maxmind-db/source-data/
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Connection-Type-Test.json
GeoIP2-2.003000/maxmind-db/source-data/README
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-ISP-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Country-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Anonymous-IP-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Enterprise-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Domain-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-City-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Precision-City-Test.json
GeoIP2-2.003000/maxmind-db/source-data/GeoIP2-Precision-ISP-Test.json
GeoIP2-2.003000/maxmind-db/MaxMind-DB-spec.md
GeoIP2-2.003000/maxmind-db/perltidyrc
GeoIP2-2.003000/maxmind-db/README.md
GeoIP2-2.003000/maxmind-db/tidyall.ini
GeoIP2-2.003000/maxmind-db/test-data/
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-broken-pointers-24.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv4-24.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Domain-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-mixed-32.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv6-32.mmdb
GeoIP2-2.003000/maxmind-db/test-data/write-test-data.pl
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Connection-Type-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-no-ipv4-search-tree.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv6-24.mmdb
GeoIP2-2.003000/maxmind-db/test-data/README.md
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv4-32.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Enterprise-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-City-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-broken-search-tree-24.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-metadata-pointers.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv6-28.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Country-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-mixed-24.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Precision-ISP-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/maps-with-pointers.raw
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-City-Test-Broken-Double-Format.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-decoder.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-string-value-entries.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Anonymous-IP-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-ISP-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/GeoIP2-Precision-City-Test.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-mixed-28.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-ipv4-28.mmdb
GeoIP2-2.003000/maxmind-db/test-data/MaxMind-DB-test-nested.mmdb
Configuring GeoIP2-2.003000 ... Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for GeoIP2
Writing MYMETA.yml and MYMETA.json
OK
Building and testing GeoIP2-2.003000 ... cp lib/GeoIP2/Role/HasLocales.pm blib/lib/GeoIP2/Role/HasLocales.pm
cp lib/GeoIP2/Record/RepresentedCountry.pm blib/lib/GeoIP2/Record/RepresentedCountry.pm
cp lib/GeoIP2/Error/Type.pm blib/lib/GeoIP2/Error/Type.pm
cp lib/GeoIP2/Role/Model.pm blib/lib/GeoIP2/Role/Model.pm
cp lib/GeoIP2/Model/ISP.pm blib/lib/GeoIP2/Model/ISP.pm
cp lib/GeoIP2/Role/Error/HTTP.pm blib/lib/GeoIP2/Role/Error/HTTP.pm
cp lib/GeoIP2/Role/HasIPAddress.pm blib/lib/GeoIP2/Role/HasIPAddress.pm
cp lib/GeoIP2/Record/Country.pm blib/lib/GeoIP2/Record/Country.pm
cp lib/GeoIP2/Model/AnonymousIP.pm blib/lib/GeoIP2/Model/AnonymousIP.pm
cp lib/GeoIP2/Model/Country.pm blib/lib/GeoIP2/Model/Country.pm
cp lib/GeoIP2/Model/Insights.pm blib/lib/GeoIP2/Model/Insights.pm
cp lib/GeoIP2/Role/Record/Country.pm blib/lib/GeoIP2/Role/Record/Country.pm
cp lib/GeoIP2/Record/Traits.pm blib/lib/GeoIP2/Record/Traits.pm
cp lib/GeoIP2/Model/Domain.pm blib/lib/GeoIP2/Model/Domain.pm
cp lib/GeoIP2/Record/MaxMind.pm blib/lib/GeoIP2/Record/MaxMind.pm
cp lib/GeoIP2/Role/Model/HasSubdivisions.pm blib/lib/GeoIP2/Role/Model/HasSubdivisions.pm
cp lib/GeoIP2/Database/Reader.pm blib/lib/GeoIP2/Database/Reader.pm
cp lib/GeoIP2/Role/Model/Location.pm blib/lib/GeoIP2/Role/Model/Location.pm
cp lib/GeoIP2/Error/WebService.pm blib/lib/GeoIP2/Error/WebService.pm
cp lib/GeoIP2/Error/IPAddressNotFound.pm blib/lib/GeoIP2/Error/IPAddressNotFound.pm
cp lib/GeoIP2/Record/Location.pm blib/lib/GeoIP2/Record/Location.pm
cp lib/GeoIP2.pm blib/lib/GeoIP2.pm
cp lib/GeoIP2/Model/Enterprise.pm blib/lib/GeoIP2/Model/Enterprise.pm
cp lib/GeoIP2/Role/Model/Flat.pm blib/lib/GeoIP2/Role/Model/Flat.pm
cp lib/GeoIP2/Model/ConnectionType.pm blib/lib/GeoIP2/Model/ConnectionType.pm
cp lib/GeoIP2/WebService/Client.pm blib/lib/GeoIP2/WebService/Client.pm
cp lib/GeoIP2/Role/Record/HasNames.pm blib/lib/GeoIP2/Role/Record/HasNames.pm
cp lib/GeoIP2/Types.pm blib/lib/GeoIP2/Types.pm
cp lib/GeoIP2/Error/Generic.pm blib/lib/GeoIP2/Error/Generic.pm
cp lib/GeoIP2/Record/Postal.pm blib/lib/GeoIP2/Record/Postal.pm
cp lib/GeoIP2/Record/City.pm blib/lib/GeoIP2/Record/City.pm
cp lib/GeoIP2/Record/Subdivision.pm blib/lib/GeoIP2/Record/Subdivision.pm
cp lib/GeoIP2/Record/Continent.pm blib/lib/GeoIP2/Record/Continent.pm
cp lib/GeoIP2/Error/HTTP.pm blib/lib/GeoIP2/Error/HTTP.pm
cp lib/GeoIP2/Model/City.pm blib/lib/GeoIP2/Model/City.pm
cp bin/web-service-request blib/script/web-service-request
"/usr/local/perlbrew/perls/stable/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/web-service-request
PERL_DL_NONLAZY=1 "/usr/local/perlbrew/perls/stable/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t t/GeoIP2/*.t t/GeoIP2/Database/*.t t/GeoIP2/Error/*.t t/GeoIP2/Model/*.t t/GeoIP2/WebService/*.t
t/00-report-prereqs.t ....................... ok
t/author-00-compile.t ....................... skipped: these tests are for testing by the author
t/author-eol.t .............................. skipped: these tests are for testing by the author
t/author-no-tabs.t .......................... skipped: these tests are for testing by the author
t/author-pod-spell.t ........................ skipped: these tests are for testing by the author
t/author-pod-syntax.t ....................... skipped: these tests are for testing by the author
t/author-test-version.t ..................... skipped: these tests are for testing by the author
t/GeoIP2/Database/Reader-Anonymous-IP.t ..... ok
t/GeoIP2/Database/Reader-Connection-Type.t .. ok
t/GeoIP2/Database/Reader-Domain.t ........... ok
t/GeoIP2/Database/Reader-Enterprise.t ....... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/8 subtests 
t/GeoIP2/Database/Reader-ISP.t .............. ok
t/GeoIP2/Database/Reader.t .................. ok
t/GeoIP2/Error/Type.t ....................... ok
t/GeoIP2/Model/City.t ....................... ok
t/GeoIP2/Model/Country.t .................... ok
t/GeoIP2/Model/Insights.t ................... ok
t/GeoIP2/Model/names.t ...................... ok
t/GeoIP2/Types.t ............................ ok
t/GeoIP2/WebService/Client.t ................ ok
t/release-cpan-changes.t .................... skipped: these tests are for release candidate testing
t/release-pod-coverage.t .................... skipped: these tests are for release candidate testing
t/release-portability.t ..................... skipped: these tests are for release candidate testing
t/release-synopsis.t ........................ skipped: these tests are for release candidate testing
t/release-tidyall.t ......................... skipped: these tests are for release candidate testing

Test Summary Report
-------------------
t/GeoIP2/Database/Reader-Enterprise.t     (Wstat: 256 Tests: 8 Failed: 1)
  Failed test:  4
  Non-zero exit status: 1
Files=25, Tests=540,  4 wallclock secs ( 0.12 usr  0.07 sys +  2.79 cusr  1.38 csys =  4.36 CPU)
Result: FAIL
FAIL 
```

**Comment by oschwald** [comment]
Thanks for reporting this. We noticed the failures on CPAN Testers. It looks like a float comparison issue on some Perl installs.

**Comment by zdm** [comment]
Tests are fails only on perl, compiled with -D uselongdouble.
Also, does this module really requires heavy Code::TidyAll dependency in production?

**Comment by oschwald** [comment]
Thanks. We also notice that the issue seemed to be limited to systems using long double. We haven't yet had a chance to determine if it is a real issue or just an issue with the tests.

The Code::TidyAll in the tests deps issue was fixed in a4065b2968f2351ef9ab4467ffdd777557c3dedb.
